idea/src/org/jetbrains/kotlin/idea/editor/wordSelection/KotlinCallExpressionWithLambdaSelectioner.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2010-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jetbrains.kotlin.idea.editor.wordSelection
+
+import com.intellij.codeInsight.editorActions.ExtendWordSelectionHandlerBase
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtCallExpression
+import org.jetbrains.kotlin.psi.KtLambdaArgument
+import org.jetbrains.kotlin.psi.psiUtil.endOffset
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+
+class KotlinCallExpressionWithLambdaSelectioner : ExtendWordSelectionHandlerBase() {
+
+    override fun canSelect(e: PsiElement): Boolean
+            = e is KtCallExpression && e.lastArgumentIsLambda() && e.lambdaIsOutsideBrackets()
+
+    override fun select(e: PsiElement?, editorText: CharSequence?, cursorOffset: Int, editor: Editor?): List<TextRange>? {
+        if (e !is KtCallExpression) return null
+
+        val endOffset = e.valueArgumentList?.endOffset ?: return null
+        return listOf(TextRange(e.startOffset, endOffset))
+    }
+
+    private fun KtCallExpression.lastArgumentIsLambda(): Boolean
+            = valueArguments.isNotEmpty() && valueArguments.last() is KtLambdaArgument
+
+    private fun KtCallExpression.lambdaIsOutsideBrackets(): Boolean
```

Comment:
This is probably unneeded. If something is parsed as `KtLambdaArgument` it won't be in `valueArgumentList`.

Reply:
Are you sure about that? If I remember correctly from debugging this in the beginning, the behaviour is as follows:

If a lambda is inside the brackets, it's part of valueArgumentList:
` foo( { "my lambda" } )`

If it's outside, it's not in valueArgumentList:
```
foo {
    "my lambda"
}
```

Might be wrong about this though, can't check it right now.

Reply:
You're right about including/excluding logic for `valueArgumentList`, but it's still enough to check that `KtLambdaArgument` is present.

`KtValueArgumentList` contains only nodes with type `VALUE_ARGUMENT`, and if there's a `KtLambdaArgument` psi element, it means that there's `LAMBDA_AGRUMENT`, and lists sizes would differ. 

Please take a look to `KtCallExpression.getValueArguments()` implementation. 

`https://github.com/JetBrains/kotlin/blob/78e278ec4c2a6c012897910d3829eb0ddead65ab/compiler/frontend/src/org/jetbrains/kotlin/psi/KtCallExpression.java#L78`

Reply:
Ah, got it.. Makes perfect sense, thanks :)